pluto_build/sql/update_empty_coord.sql

```diff
@@ -1,34 +1,24 @@
 -- Update X/Y and Lat/Long fields that are NULL with values from Lot Centroids
 -- Make sure lot centroids fall within the polygon
-SELECT bbl, centroid, point_surface,
-        _ST_CONTAINS(geom, ST_SetSRID(centroid ,4326)) as contain
-INTO update_table
-FROM(SELECT p.bbl as bbl, p.geom as geom, ST_Centroid(p.geom) as centroid, 
-	 		ST_PointOnSurface(p.geom) as point_surface
+WITH update_table AS
+(
+SELECT bbl, 
+	CASE -- check whether the centroid locates in the polygon, otherwise use the PointOnSurface function
+		WHEN _ST_CONTAINS(geom, ST_SetSRID(centroid ,4326)) IS TRUE THEN centroid
```

Comment:
Nice! This seems neater to me. I have one small comment on field naming: I don't know if the second field in update table should be called `centroid` as sometimes it's technically not a centroid. I think it makes more sense to call it something like `polygon_center` or something like that. 
I unfortunately messed up my local build and I'm re-building pluto now so someone with a working local build should test the code itself

Reply:
Ok if you think that's best we can keep it